spec/rubocop/cop/lint/out_of_range_regexp_ref_spec.rb

```diff
@@ -234,6 +234,46 @@
       end
     end
 
+    context 'matching hashes' do
+      it 'uses the maximum number of captures with multiple patterns' do
+        expect_no_offenses(<<~RUBY)
+          case hash
+          in a: /(foo)(bar)/
+            $2
+          end
+        RUBY
+
+        expect_offense(<<~RUBY)
+          case hash
+          in a: /(foo)(bar)/, b: /(bar)baz/
+            $3
+            ^^ $3 is out of range (2 regexp capture groups detected).
+          end
+        RUBY
+      end
```

Comment:
Can you make `expect_no_offenses` and `expect_offense` different `it` for test case isolation?